torch/_dynamo/variables/builder.py

```diff
@@ -853,6 +856,35 @@ def wrap_tensor(self, value: torch.Tensor):
 
         return tensor_variable
 
+    def wrap_numpy_ndarray(self, value):
```

Comment:
This is a kind of a strange thing to add. While on its own it makes sense (it makes it possible for us to compile code that is directly calling `torch_np.ndarray`), it is not clear to me why you would need it to implement graph break. In particular, I would expect the graph breaks to generate NUMPY ndarray, not torch ndarrays.

It is also not clear to me if you even need this if someone is directly using torch_np.ndarray; you don't need to treat it directly as a TensorVariable, it can just be a user defined struct that contains a tensor, assuming we have enough inlining going on the outer wrapper should just evaporate.

Reply:
So, when we graph break we will compile the subgraph. Let's say the subgraph has `<tensor>.numpy()` calls, we already wire torch_np to those calls and they are actually torch_np.ndarrays. In this method I'm just wrapping these torch_np.ndarrays in NumpyNdarrayVariable so they can be used later. If I don't do this they got wrapped into NumpyVariable which is wrong.

I'm not sure what's going on with the example_value but seems like dynamo doesn't take a None for that. I'll see if I can clean this up a bit

Reply:
I'm saying, the torch_np.ndarray shouldn't be showing up at all when you wrap. You should only ever see numpy.ndarray and torch.Tensor from userland.